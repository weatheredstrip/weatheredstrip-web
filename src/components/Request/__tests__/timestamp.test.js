import React from 'react'
import { render, cleanup } from '@testing-library/react'
import Timestamp from '../timestamp.react'
import moment from 'moment'
import '@testing-library/jest-dom/extend-expect'
afterEach(cleanup)

it('timestamp titles render properly', () => {
  const sometime = new Date('2018-08-15')
  const formattedSometime = moment
    .utc(sometime)
    .format('YYYY-MM-DD[T]HH:mm:ss[Z]')
  const { container, getByText } = render(<Timestamp dataTime={sometime} />)
  const timestamp = container.firstChild

  const timestampTitles = timestamp.firstChild
  expect(timestampTitles).toMatchSnapshot()
})

it('timestamp times render properly', () => {
  const sometime = new Date('2018-08-15')
  const formattedSometime = moment
    .utc(sometime)
    .format('YYYY-MM-DD[T]HH:mm:ss[Z]')
  const { container, getByText } = render(<Timestamp dataTime={sometime} />)
  const timestamp = container.firstChild
  expect(
    getByText(moment.utc().format('YYYY-MM-DD[T]HH:mm:ss[Z]')),
  ).toBeInTheDocument()
  expect(getByText(formattedSometime)).toBeInTheDocument()
})
