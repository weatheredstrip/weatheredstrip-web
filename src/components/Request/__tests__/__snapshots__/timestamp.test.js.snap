// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`timestamp titles render properly 1`] = `
<div
  class="timestamp-names"
>
  <div>
    Data time:
  </div>
  <div>
    Current time:
  </div>
</div>
`;
